packages/@aws-cdk/aws-lambda-nodejs/README.md

```diff
@@ -0,0 +1,56 @@
+## Amazon Lambda Node.js Library
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> **This is a _developer preview_ (public beta) module. Releases might lack important features and might have
+> future breaking changes.**
+>
+> This API is still under active development and subject to non-backward
+> compatible changes or removal in any future version. Use of the API is not recommended in production
+> environments. Experimental APIs are not subject to the Semantic Versioning model.
+
+---
+<!--END STABILITY BANNER-->
+
+This library provides constructs for Node.js Lambda functions.
+
+### Node.js Function
+Define a `NodejsFunction`:
+
+```ts
+new lambda.NodejsFunction(this, 'my-handler');
+```
+
+By default, the construct will look up for a `index.ts` or `index.js` that exports a `handler`
+function in the folder named as the construct's id:
+```
+.
+├── my-handler
+|   └── index.ts # exports a function named 'handler'
+├── stack.ts # defines a 'NodeJsFunction' with 'my-handler' as id
+```
+
+This file is used as "entry" for [Parcel](https://parceljs.org/). This means that your code is
+automatically transpiled and bundled whether it's written in JavaScript or TypeScript.
+
+Alternatively, an entry file and handler can be specified:
+```ts
+new lambda.NodejsFunction(this, 'MyFunction', {
+  entry: '/path/to/my/file.ts'
+  handler: 'myExportedFunc'
+});
+```
+
+All other properties of `lambda.Function` are supported, see also the [AWS Lambda construct library](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda).
+
+### Configuring Parcel
+The `NodejsFunction` construct exposes some Parcel options via properties: `minify`, `sourceMaps`,
+`buildDir` and `cacheDir`.
+
+Parcel transpiles your code (every internal module) with `@babel/preset-env` and uses the
+`engines.node` defined in `package.json` as target (defaults to Node.js 8).
+
+Configuring Babel with Parcel is possible via a `.babelrc` or a `babel` config in `package.json`.
```

Comment:
I also think it should be possible to configure these through the construct (probably a better practice).

Reply:
This is the only way to configure Babel when using Parcel (no CLI option). So we  would have to write a `.babelrc` at the root of the package before calling `parcel`... not really fan of this

Reply:
Can we write a babelrc file in the .build directory and then remove it at the end? I do think it makes more sense that all config will be in code.

Reply:
What if this config already exists (either in `.babelrc` or `package.json`)?

Reply:
without a pre-existing .babelrc, parcel already does some base level transpilation it seems. Does that mean to user only has to define a `.babelrc` if they want babel config beyond the default? That's the ideal setup in my mind. Most users that are using babel and know about it are used to `.babelrc` and it feels like the idiomatic way to do config to me.

From parcel docs
> In addition to any transforms specified in .babelrc, Parcel always uses Babel on all modules to compile modern JavaScript into a form supported by browsers. See the JavaScript/Default Babel Transforms section for more information.

Reply:
> Does that mean to user only has to define a .babelrc if they want babel config beyond the default?

Yes